hawc/apps/assessment/views.py

```diff
@@ -448,31 +449,105 @@ class AttachmentCreate(BaseCreate):
     success_message = "Attachment added."
     parent_model = models.Assessment
     parent_template_name = "parent"
+    template_name = "assessment/components/attachment_edit_row.html"
     model = models.Attachment
-    form_class = forms.AttachmentForm
+    form_class = forms.NewAttachmentForm
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context["newAttach"] = True
+        return context
+
+    @transaction.atomic
+    def form_valid(self, form):
+        super().form_valid(form)
+        context = self.get_context_data()
+        context["object_list"] = models.Attachment.objects.get_attachments(
+            self.assessment, not context["obj_perms"]["edit"]
+        )
+        context["canEdit"] = context["obj_perms"]["edit"]
+        # context["newAttach"] = True
+        return render(self.request, "assessment/components/attachment_row.html", context)
 
 
 class AttachmentRead(BaseDetail):
     model = models.Attachment
 
     def get(self, request, *args, **kwargs):
+        if request.GET.get("test", -1) != -1:
```

Comment:
it might be easier to check if the header is htmx and if so return the template, i think htmx requests use a header which would be nice to have an is_htmx utility function